src/coreclr/utilcode/stgpooli.cpp

```diff
@@ -212,7 +212,7 @@ HRESULT CPackedLen::SafeGetData(    // S_OK, or error
 
     BYTE const *pbData = reinterpret_cast<BYTE const*>(*ppData);
 
-    if (pbData + *pcbData < pbData)
+    if (*pcbData > reinterpret_cast<SIZE_T>(pDataSourceEnd) - reinterpret_cast<SIZE_T>(pDataSource))
```

Comment:
with this change, I think you can remove the followon check for data buffer overflow (and move that comment to this if statement.

Reply:
I don't think so, although initially I did do that. This only checks the range of the index, not how far it indexes into the data buffer. The check below ensures that starting from `pbData`, this index doesn't exceed the overall block.